admin/class-admin-recommended-replace-vars.php

```diff
@@ -0,0 +1,161 @@
+		// Custom post.
+		'custom_post_type'        => array(),
+
+		// Taxonomies.
+		'category'                => array( 'sitename', 'title', 'sep' ),
+		'post_tag'                => array( 'sitename', 'title', 'sep' ),
+		// Specific custom taxonomies.
+		'post_format'             => array( 'sitename', 'title', 'sep', 'page' ),
+		'product_cat'             => array( 'sitename', 'title', 'sep' ),
+		'product_tag'             => array( 'sitename', 'title', 'sep' ),
+		'product_shipping_class'  => array( 'sitename', 'title', 'sep', 'page' ),
+		'product_brand'           => array( 'sitename', 'title', 'sep' ),
+		'pwb-brand'               => array( 'sitename', 'title', 'sep' ),
+		// Custom taxonomy.
+		'term-in-custom-taxomomy' => array( 'sitename', 'title', 'sep' ),
+
+		// Settings - archive pages.
+		'author_archive'          => array( 'sitename', 'title', 'sep', 'page' ),
+		'date_archive'            => array( 'sitename', 'sep', 'date', 'page' ),
+		'product_archive'         => array( 'sitename', 'sep', 'page', 'pt_plural' ),
+		'custom-taxonomy_archive' => array( 'sitename', 'title', 'sep' ),
+		// Settings - special pages.
+		'search'                  => array( 'sitename', 'searchphrase', 'sep', 'page' ),
+		'404'                     => array( 'sitename', 'sep' ),
+	);
+
+	/**
+	 * Determines the page type of the current term.
+	 *
+	 * @param string $taxonomy The taxonomy name.
+	 *
+	 * @return string The page type.
+	 */
+	public function determine_for_term( $taxonomy ) {
+		$supported_taxonomies = array( 'category', 'tag', 'post_format' );
+
+		// Add WooCommerce specific types.
+		if ( $this->is_woocommerce_active() ) {
+			$supported_taxonomies[] = 'product_cat';
+			$supported_taxonomies[] = 'product_tag';
+			$supported_taxonomies[] = 'product_brand';
+			$supported_taxonomies[] = 'product_shipping_class';
+			// This is the for the plugin: Perfect WooCommerce Brands.
+			$supported_taxonomies[] = 'pwb-brand';
+		}
+
+		if ( in_array( $taxonomy, $supported_taxonomies, true ) ) {
+			return $taxonomy;
+		}
+
+		return 'term-in-custom-taxomomy';
+	}
+
+	/**
+	 * Determines the page type of the current post.
+	 *
+	 * @param WP_Post $post The WordPress global post variable.
+	 *
+	 * @return string The page type.
+	 */
+	public function determine_for_post( $post ) {
+		if ( $post instanceof WP_Post === false ) {
+			return 'post';
+		}
+
+		if ( $post->post_type === 'page' && $this->is_homepage( $post ) ) {
+			return 'homepage';
+		}
+
+		$supported_post_types = array( 'post', 'page' );
+
+		// Add WooCommerce specific types.
+		if ( $this->is_woocommerce_active() ) {
+			$supported_post_types[] = 'product';
+		}
+
+		if ( in_array( $post->post_type, $supported_post_types, true ) ) {
+			return $post->post_type;
+		}
+
+		return 'custom_post_type';
+	}
+
+	/**
+	 * Retrieves the recommended replacement variables for the given post type.
+	 *
+	 * @param string $post_type The post type.
```

Comment:
Can you rename this to `page_type`

Reply:
We called this page_type earlier. I changed it to post type because that is how WordPress calls it.